nncf/onnx/quantization/quantize_model.py

```diff
@@ -44,6 +54,60 @@
 TTensor = TypeVar("TTensor")
 
 
+def check_model_protobuf_size(model: onnx.ModelProto) -> None:
+    """
+    Checks whether the serialized ONNX model exceeds the 2GB protobuf size limit.
+    If the size limit is exceeded, a `nncf.ValidationError` is raised.
+
+    :param model: The ONNX model to be checked.
+    """
+    MAXIMUM_PROTOBUF = 2000000000  # Limitation of single protobuf file is 2GB
+    protobuf_string = model.SerializeToString()
+    if sys.getsizeof(protobuf_string) > MAXIMUM_PROTOBUF:
+        msg = (
+            "The protobuf of onnx model is too large (>2GB). "
+            "Please load the model with the `load_external_data` flag set to `False`. "
+            "For more details, please visit: https://onnx.ai/onnx/repo-docs/ExternalData.html"
+        )
+        raise nncf.ValidationError(msg)
+
+
+def check_external_data_location(model: onnx.ModelProto, external_data_dir: Optional[str] = None) -> None:
+    """
+    Raises `nncf.ValidationError` if any referenced external data file does not exist, is not a regular file,
+    or is a symlink.
+
+    :param model: The ONNX model to validate.
+    :param external_data_dir: Path to the directory where the external data files are expected to be located.
+        If None, the current working directory is used.
+    """
+    # If external_data_dir is not provided, we should test against the current working directory.
+    external_data_dir = Path.cwd() if external_data_dir is None else Path(external_data_dir)
+
+    if not external_data_dir.is_absolute():
+        msg = (
+            f"BackendParameters.EXTERNAL_DATA_DIR should be an absolute path, but {str(external_data_dir)} "
+            "was provided instead."
+        )
+        raise nncf.ValidationError(msg)
+
+    for tensor in _get_all_tensors(model):
+        if uses_external_data(tensor):
+            info = ExternalDataInfo(tensor)
+            # `info.location` field stores file path relative to the filesystem directory where
+            # the ONNX protobuf model was stored. Note, that up-directory path components such
+            # .. are disallowed and should be stripped when parsing.
+            # Source: https://onnx.ai/onnx/repo-docs/ExternalData.html
+            external_data_file_name = Path(info.location).name  # Extract only the filename
+            data_path = external_data_dir / external_data_file_name
+            if not data_path.exists() or not data_path.is_file() or data_path.is_symlink():
```

Comment:
ONNX model often has one data file. You will check the same file for each tensor, am I right? If this statement is correct, this is not efficient approach.

Reply:
Fixed